src/sidebar/components/Thread.js

```diff
@@ -73,64 +158,64 @@ function Thread({ thread, threadsService }) {
       ),
     [
       hasAppliedFilter,
+      isReply,
       onToggleReplies,
       thread.annotation,
-      thread.parent,
       thread.replyCount,
       thread.collapsed,
       thread.visible,
     ]
   );
 
   return (
-    <section
-      className={classnames('Thread', {
-        'Thread--reply': thread.depth > 0,
-        'is-collapsed': thread.collapsed,
-      })}
-    >
-      {showThreadToggle && (
-        <div className="Thread__collapse">
-          <div className="Thread__collapse-button-container">
-            <IconButton
-              classes="NonResponsiveIconButton"
-              expanded={!thread.collapsed}
-              icon={toggleIcon}
-              title={toggleTitle}
-              onClick={onToggleReplies}
-              size="medium"
-              variant="light"
-            />
-          </div>
-        </div>
+    <section className="flex" data-testid="thread-container">
+      {isReply && (
+        <ThreadCollapseControl
+          threadIsCollapsed={thread.collapsed}
+          onToggleReplies={onToggleReplies}
+        />
       )}
 
-      <div className="Thread__content">
+      <div
+        className={classnames(
+          // Set a max-width to ensure that annotation content does not exceed
+          // the width of the container
+          'grow max-w-full'
+        )}
+        data-testid="thread-content"
+      >
         {annotationContent}
 
-        {showHiddenToggle && (
-          <>
+        {isHidden && (
+          <div className="space-y-2">
             {!thread.parent && (
-              <ThreadHeader
+              <HiddenThreadCardHeader
```

Comment:
If you move the `isEditing` logic into `AnnotationHeader` then you could skip the helper and just render `thread.parent && annotation && <AnnotationHeader .../>` here.